elk/utils/types.py

```diff
@@ -0,0 +1,11 @@
+from enum import Enum
+
+
+class PromptEnsembling(Enum):
```

Comment:
Did we really pick a good name here? Is PromptEnsembling a good name?

Reply:
It's kind of bad but I don't really know better ones

Reply:
Well it's bad because per the comments on `discord:EleutherAI:eliciting-latent-knowledge`
```
full = all the ensembling (over contrast pair and templates)
partial = ensembling only over contrast pair (averaging p+ and 1 - p-)
none = looking only at p(input) according to probe, which is most likely what people would do at inference time to keep it simple
If you want you could also define the fourth ensembling method for ensembling over just the templates but not the contrast pair, but that's not how we defined it (we left that option out)
```
It's not only prompt ensembling that this variable controls. It's also pooling over the contrast pair, maybe warranting a more generic name